joinmarket/blockchaininterface.py

```diff
@@ -115,6 +115,341 @@ def estimate_fee_per_kb(self, N):
+                        if res == False:
+                            continue
+                        data_tx = json.loads(res)
+                        res = BlockchaininfoInterface().make_bci_request(bcinfo_url + tx + '?format=hex')
+                        data_tx['hex'] = str(res)
+                        data_list.append(data_tx)
+
+                    if not isinstance(data_list, list):
+                        data_list = [data_list]
+
+                    for txinfo in data_list:
+                        txhex = str(txinfo['hex'])
+                        outs = set([(sv['script'], sv['value'])
+                                    for sv in btc.deserialize(txhex)['outs']])
+                        log.debug('unconfirm query outs = ' + str(outs))
+                        if outs == self.tx_output_set:
+                            unconfirmed_txid = txinfo['hash']
+                            unconfirmed_txhex = str(txinfo['hex'])
+                            break
+
+                self.unconfirmfun(
+                        btc.deserialize(unconfirmed_txhex), unconfirmed_txid)
+
+                st = int(time.time())
+                confirmed_txid = None
+                confirmed_txhex = None
+                bcinfo_url = 'https://blockchain.info/rawtx/'
+
+                while not confirmed_txid:
+                    time.sleep(confirm_poll_period)
+                    if int(time.time()) - st > confirm_timeout:
+                        log.debug('checking for confirmed tx timed out')
+                        return
+                    
+                    for tx in shared_txid:
+                        res = btc.make_request(bcinfo_url + tx)
+                        data = json.loads(res)
+
+                        # Not yet confirmed
+                        if 'block_height' not in data:
+                            continue
+                        else:
+                            res = BlockchaininfoInterface().make_bci_request(bcinfo_url + tx + '?format=hex')
+                            txhex = str(res)
+                            outs = set([(sv['script'], sv['value'])
+                                    for sv in btc.deserialize(txhex)['outs']])
+                            log.debug('confirm query outs = ' + str(outs))
+                            if outs == self.tx_output_set:
+                                confirmed_txid = txinfo['hash']
+                                confirmed_txhex = str(txinfo['hex'])
+                                break
+
+                self.confirmfun(
+                        btc.deserialize(confirmed_txhex), confirmed_txid, 1)
+
+        NotifyThread(txd, unconfirmfun, confirmfun).start()
+
+    def pushtx(self, txhex):
+        try:
+            json_str = btc.bci_pushtx(txhex)
+        except Exception:
+            log.debug('failed blockchain.info pushtx')
+            return None
+
+        # This part is adapted from BlockrInterface pushtx()
```

Comment:
I'd feel more comfortable looking up the existence of the tx in the network before proceeding to make sure the push succeeded, but I am admitting the approach in pushtx() is far from perfect.